testplan/web_ui/testing/src/index.js

```diff
@@ -17,10 +18,21 @@ import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 const AppRouter = () => (
   <Router>
     <Switch>
-      <Route path="/testplan/:uid/:selection*" component={BatchReport} />
+      <Route
+        path="/testplan/:uid/:selection*"
+        render={({match}) => (
+          <ErrorBoundary>
+            <BatchReport match={match}/>
+          </ErrorBoundary>
+        )}
+      />
       <Route
         path="/interactive/:uid?/:selection*"
-        component={InteractiveReport}
+        render={({match}) => (
+          <ErrorBoundary>
+            <InteractiveReport match={match}/>
+          </ErrorBoundary>
```

Comment:
QQ: isn't it enough to wrap the BatchReport and InteractiveReport? If error already leads out of the component, my naive guess is that any lower level error would propagate up to this very point and these two ErrorBoundary instances would take care of everything.

Reply:
Already discussed offline, but putting here as well.
We want to catch render errors in the lowest component level possible (where it makes sense) to keep the functionality of the rest of the page. If we only wrap the BatchReport for example, a render failure in the Assertion level will prevent the user to access every other AssertionGroups.